### PR TITLE
2384 unavailability

### PR DESCRIPTION
**Pull request by atGit2021**
Hi Daniel, I'm creating this PR for your review.  It contains the unavailability.pg.repository file along with the CRUD operations to work with postgres db for the migration effort.  Thank you, Andre

┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/3466557237) by [Unito](https://www.unito.io)

### CODE REVIEW

File: src/components/user/unavailability/unavailability.pg.repository.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    if (!rows) {
+      throw new ServerException('Failed to create unavailability');
+    }
+    return rows[0].id;
+  }
+
+  async readOne(id: ID): Promise<UnsecuredDto<Unavailability>> {
+    const rows = await this.pg.query<UnsecuredDto<Unavailability>>(
+      `
+      SELECT id, admin_people_id, period_start, period_end, description,
```
Comment (daniel-leod) [inline src/components/user/unavailability/unavailability.pg.repository.ts]: According to the `Unavailability` dto we only need `description`, `start` and `end` fields. Please omit the next columns: `admin_people_id`, `created_by_admin_people_id as "creator"`, `modified_at as "modifiedAt"`, `modified_by_admin_people_id`

File: src/components/user/unavailability/unavailability.pg.repository.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    return rows[0];
+  }
+
+  async readMany(
+    ids: readonly ID[]
+  ): Promise<ReadonlyArray<UnsecuredDto<Unavailability>>> {
+    const rows = await this.pg.query<UnsecuredDto<Unavailability>>(
+      `
+      SELECT id, admin_people_id, period_start, period_end, description, 
+      created_at as "createdAt", created_by_admin_people_id as "creator",
```
Comment (daniel-leod) [inline src/components/user/unavailability/unavailability.pg.repository.ts:67]: Apply those changes mentioned before here too.

File: src/components/user/unavailability/unavailability.pg.repository.ts [hunk 3]
```diff
@@ -74,6 +74,48 @@ export class UnavailabilityPgRepository extends DtoRepository(Unavailability) {
...
+    const limit = input.count;
+    const offset = (input.page - 1) * input.count;
+    const [{ count }] = await this.pg.query<{ count: string }>(
+      'SELECT count(*) FROM sc.person_unavailabilities;'
+    );
+
+    const rows = await this.pg.query<UnsecuredDto<Unavailability>>(
+      `
+      SELECT id, admin_people_id, period_start as "start", period_end as "end", description, 
+      created_at as "createdAt", created_by_admin_people_id as "creator",
```
Comment (daniel-leod) [inline src/components/user/unavailability/unavailability.pg.repository.ts:105]: Same here.

**Review by daniel-leod — CHANGES_REQUESTED** [review]

File: src/components/user/unavailability/unavailability.pg.repository.ts (hunk 1 above)
Comment (atGit2021) [reply src/components/user/unavailability/unavailability.pg.repository.ts]: just pushed commit [e3bc2fb] that removes the unnecessary fields

File: src/components/user/unavailability/unavailability.pg.repository.ts (hunk 2 above)
Comment (atGit2021) [reply src/components/user/unavailability/unavailability.pg.repository.ts]: just pushed commit [e3bc2fb] that removes the unnecessary fields

File: src/components/user/unavailability/unavailability.pg.repository.ts (hunk 3 above)
Comment (atGit2021) [reply src/components/user/unavailability/unavailability.pg.repository.ts]: just pushed commit [e3bc2fb] that removes the unnecessary fields

**Review by daniel-leod — APPROVED** [review]

**Comment by CarsonF** [comment]
Closing postgres migration PRs for now until a more wholistic strategy/plan can be formed.

**Pull request closed without merging**